### PR TITLE
SQL-1168: Simplfiy Anyof(<one>) and double buffer

### PR DESCRIPTION
**Pull request by terakilobyte**
This PR simplifies schemas that are AnyOf one concrete type that we have the ability to show.

### CODE REVIEW

File: integration_test/tests/query_tests.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 mod common;
 
+#[cfg(target_os = "linux")]
```
Comment (terakilobyte) [inline integration_test/tests/query_tests.rs:3]: This test suite routinely segfaults on Windows. Attempting to troubleshoot this with @pmeredit led us to deciding to disable the test for Windows for now.

File: core/src/json_schema.rs [hunk 2]
```diff
@@ -232,12 +233,21 @@ pub mod simplified {
...
+                    // if there is an AnyOf that contains one item, we can simplify it
+                    if any_of.len() == 1 {
+                        return Schema::try_from(any_of.into_iter().next().unwrap());
+                    }
+                    Ok(Schema::AnyOf(
+                        any_of
+                            .into_iter()
+                            .map(Atomic::try_from)
+                            .collect::<Result<BTreeSet<Atomic>>>()?,
+                    ))
```
Comment (pmeredit) [inline core/src/json_schema.rs:249]:
```suggestion
                    match any_of.len() {
                        0 => Err(Error::InvalidResultSetJsonSchema),
                         // if there is an AnyOf that contains one item, we can simplify it
                        1 => Schema::try_from(any_of.into_iter().next().unwrap()),
                        _ => Ok(Schema::AnyOf(
                            any_of
                                .into_iter()
                                .map(Atomic::try_from)
                                .collect::<Result<BTreeSet<Atomic>>>()?,
                        ))
                    }       
```

Comment (pmeredit) [reply core/src/json_schema.rs:249]: This cleans up the code a bit, imo.

**Review by pmeredit — CHANGES_REQUESTED** [review]
Also, the double buffer change isn't in here. Perhaps this ticket should just be simplifying AnyOf anyway?

File: core/src/json_schema.rs (hunk 2 above)
Comment (terakilobyte) [reply core/src/json_schema.rs:249]: agreed, thanks for the suggesion!

**Comment by terakilobyte** [comment]
@pmeredit Yeah we (@nbagnard and I) decided to remove that, as looking at it more closely we identified some issues with number of chars vs number of bytes, and decided to split that out into separate work.

**Review by nbagnard — APPROVED** [review]
LGTM! 👍

**Review by pmeredit — APPROVED** [review]
LGTM! Just make sure to change the commit message

**Comment by terakilobyte** [comment]
evergreen merge

**Comment by pmeredit** [comment]
evergreen merge

**Pull request merged**